tests/Doctrine/Tests/ORM/Functional/Ticket/GH9027Test.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\ORM\UnitOfWork;
+use Doctrine\Tests\OrmFunctionalTestCase;
+use Exception;
+
+class GH9027Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        try {
+            $this->_schemaTool->createSchema(
+                [
+                    $this->_em->getClassMetadata(GH9027Customer::class),
+                    $this->_em->getClassMetadata(GH9027Cart::class),
+                ]
+            );
+        } catch (Exception $e) {
+        }
+    }
+
+    /**
+     * @group GH-9027
+     */
+    public function testUnitOfWorkHandlesNullRelations(): void
+    {
+        $uow = new UnitOfWork($this->_em);
+        $hints = ['fetchMode' => [GH9027Cart::class => ['customer' => ClassMetadata::FETCH_EAGER]]];
+        $cart = $uow->createEntity(GH9027Cart::class, ['id' => 1, 'customer' => 24252], $hints);
```

Comment:
```suggestion
        $cart = $uow->createEntity(
                GH9027Cart::class,
                ['id' => 1, 'customer' => 24252],
                ['fetchMode' => [GH9027Cart::class => [
                        'customer' => ClassMetadata::FETCH_EAGER
                ]]];
        );
```

Reply:
I would've preferred this too, but `$hints` needs to be passed by reference, not value.

Reply:
O_o OK